ui/modules/auth/inscription/InscriptionStep2.jsx

```diff
@@ -0,0 +1,170 @@
+import React from "react";
+import { useFormik } from "formik";
+import { useRouter } from "next/router";
+import * as Yup from "yup";
+import generator from "generate-password-browser";
+import {
+  Box,
+  Button,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  HStack,
+  Input,
+  Text,
+  RadioGroup,
+  Radio,
+  Heading,
+} from "@chakra-ui/react";
+import { _post } from "@/common/httpClient";
+import Ribbons from "@/components/Ribbons/Ribbons";
+import { SIRET_REGEX } from "@/common/domain/siret";
+import useToaster from "@/hooks/useToaster";
+
+const InscriptionStep2 = ({ onSucceeded, type, uai, organismesAppartenance, etablissement, ...props }) => {
+  const { toastError } = useToaster();
+  const router = useRouter();
+  const uaiToDisplay = etablissement.uai || uai || undefined;
+
+  const { values, handleChange, handleSubmit, errors, touched, setErrors, isValid, isValidating } = useFormik({
+    initialValues: {
+      email: "",
+      nom: "",
+      civility: "",
+      prenom: "",
+      type,
+      organismes_appartenance: organismesAppartenance.toUpperCase(),
+      siret: etablissement.siret,
+      uai,
+    },
+    validationSchema: Yup.object().shape({
+      type: Yup.string().required("Requis"),
+      email: Yup.string().email("Format d'email invalide").required("Votre email est obligatoire"),
+      siret: Yup.string()
+        .matches(SIRET_REGEX, {
+          message: "SIRET invalide",
+          excludeEmptyString: true,
+        })
+        .required("Le siret est obligatoire"),
+      uai: Yup.string(),
+      organismes_appartenance: Yup.string().required("organismes_appartenance obligatoire"),
+      nom: Yup.string().required("Votre nom est obligatoire"),
+      civility: Yup.string().required("Votre civilité est obligatoire"),
+      prenom: Yup.string().required("Votre prénom est obligatoire"),
+    }),
+    onSubmit: async (values) => {
+      // eslint-disable-next-line no-undef, no-async-promise-executor
+      try {
+        const newTmpPassword = generator.generate({
+          length: 12,
+          numbers: true,
+          lowercase: true,
+          uppercase: true,
+          strict: true,
+        });
+        const result = await _post("/api/v1/auth/register", { ...values, password: newTmpPassword });
```

Comment:
HS: j'aurais bien vu ce mot de passe temporaire géré côté API plutôt... :upside_down_face:

Reply:
+1 mais out of scope